docs/adr/0000-use-es-compatible-unit-test-frameworks.md

```diff
@@ -0,0 +1,174 @@
+
+## Considered Options
+
+1. Transpile code under test to ES5 using Babel, and continue with Jest
+2. Use an ES6+ compatible test runner, and stop using mocks
+3. Use an ES6+ compatible test runner, along with an ES6+ compatible mocking library
+
+## Decision Outcome
+
+Chosen option: 3. Use an ES6+ compatible test runner, along with an ES6+ compatible mocking library.
+
+While we should strive to reduce our use of test doubles, we have quite a few existing tests, and we should also try to
+avoid reducing test coverage until we refactor the codebase to use other approaches to make the code more testable
+without using mocks/stubs/fakes.
+
+We've chosen the following tools:
+
+* [AVA](https://github.com/avajs/ava) as a test runner
+* [testdouble.js aka td.js](https://github.com/testdouble/testdouble.js/) as a mock library
+  * Optionally uses [quibble](https://github.com/testdouble/quibble) for ES module interception
+
+Both of these are subject to change, but the fulfill the basic requirements and require fairly minimal changes
+from our current setup with Jest.
+
+### Positive Consequences
+
+* remove the need to transpile the code to older version of JS
+* current tests can be ported over with minimal changes
+* no changes needed to system under test (existing code)
+
+### Negative Consequences
+
+* module interception requires the use of [experimental module loader feature](https://nodejs.org/api/esm.html#loaders) in Node
+* current test code needs changes to support new test runner and mocking library, and dynamic imports
+
+## Pros and Cons of the Options
+
+### Jest using Babel for transpilation to ES5
+
+* Good, because both system under test and existing test code need no changes
+* Good, because no experimental features are required (node ESM support)
+* Bad, because the system under test is modified by transpilation to ES5
+* Bad, because the test code will require changes eventually (the current ES5 Jest approach will not work according to ES spec)
+
+### Use an ES6+ compatible test runner, and stop using mocks
+
+* Good, because the system under test and the test code are pure ES modules
+* Good, because no experimental features are required (Node experimental module loader)
+* Bad, because both the system under test and also unit test need changes
+  * the system under test needs to be rewritten to support being tested without mocks
+  * the unit tests need to be rewritten to use the new non-jest test runner
+
+### Use an ES6+ compatible test runner, along with an ES6+ compatible mocking library
+
+* Good, because the system under test and the test code are pure ES modules
+* Good, because the system under test needs no changes
+* Bad, because the unit tests need to be rewritten to use the new non-jest test runner and mocking library
+* Bad, because an experimental feature is required (Node experimental module loader)
```

Comment:
As above, maybe I'm confusing the feature being referenced – I understood ESM is fully supported by our version of Node?

Reply:
Sorry this is incorrect, will fix!

Reply:
Oops wait this is not incorrect - this is actually a different feature :)
`--loader` https://nodejs.org/api/esm.html#loaders

Reply:
I've clarified this as well (thanks for pointing it out, I didn't remember this detail at first when you pointed it out!)